### PR TITLE
[cinatra, cmdline] Add new ports

### PR DESCRIPTION
**Pull request by EmployeeNo427**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by EmployeeNo427** [comment]
@microsoft-github-policy-service agree

**Comment by LilyWangLL** [comment]
Please fix the CI pipeline error as below:
```
##[error]versions/c-/cinatra.json(2,): error : cinatra is missing a version database file at version/c-/cinatra.json
Run:
vcpkg x-add-version cinatra
to create the versions file.
##[error]versions/c-/cmdline.json(2,): error : cmdline is missing a version database file at version/c-/cmdline.json
Run:
vcpkg x-add-version cmdline
to create the versions file.
```

File: ports/cinatra/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "name": "cinatra",
+    "version-string": "0.9.1",
```
Comment (LilyWangLL) [inline ports/cinatra/vcpkg.json:3]:
```suggestion
    "version": "0.9.1",
```

File: ports/cinatra/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,24 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO qicosmos/cinatra
+    REF 0.9.1
```
Comment (LilyWangLL) [inline ports/cinatra/portfile.cmake:4]:
```suggestion
    REF ${VERSION}
```

File: ports/cinatra/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    SHA512 55bba51bb6190b76bc6415d3c36e82daeb5f174f1cb10490951cf05863f55653a6d68ffd3b66c5b3b2ebb6e68a7a84c090e973a6718f3f2a5849b04330e4b180
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+)
+
+vcpkg_install_cmake()
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/cinatra RENAME copyright)
```
Comment (LilyWangLL) [inline ports/cinatra/portfile.cmake:14]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

File: ports/cinatra/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+vcpkg_copy_pdbs()
+file(REMOVE_RECURSE ${SOURCE_PATH}/.git)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug)
+
+
+
+
+
+
```
Comment (LilyWangLL) [inline ports/cinatra/portfile.cmake:24]:
```suggestion
```

File: ports/cmdline/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "cmdline",
+  "version-string": "2014.02.04",
```
Comment (LilyWangLL) [inline ports/cmdline/vcpkg.json:3]:
```suggestion
  "version-date": "2014-02-04",
```

File: ports/cmdline/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "cmdline",
+  "version-string": "2014.02.04",
+  "description": "A simple, header-only command line parser for C++.",
+  "homepage": "https://github.com/tanakh/cmdline",
+  "dependencies": []
```
Comment (LilyWangLL) [inline ports/cmdline/vcpkg.json:6]:
```suggestion
  "homepage": "https://github.com/tanakh/cmdline",
  "license": "BSD-3-Clause"
```

File: ports/cmdline/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    HEAD_REF master
+)
+
+file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/cmdline)
+
+file(INSTALL ${SOURCE_PATH}/cmdline.h
+     DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+
+     
+
```
Comment (LilyWangLL) [inline ports/cmdline/portfile.cmake:15]:
```suggestion
```

File: ports/cmdline/portfile.cmake [hunk 8]
```diff
@@ -0,0 +1,15 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO tanakh/cmdline
+    REF master
+    SHA512 4d16185c9d8a75a90f21063580301a837bfbb40d9d2aae1449243d5dcbd1d32ea78701ee0c9b74b8e527cc9d36e9397023f08a9dd3d96a55acec9fbcf532a635
+    HEAD_REF master
+)
+
+file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/cmdline)
```
Comment (LilyWangLL) [inline ports/cmdline/portfile.cmake:9]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

File: ports/cmdline/portfile.cmake [hunk 9]
```diff
@@ -0,0 +1,15 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO tanakh/cmdline
+    REF master
```
Comment (LilyWangLL) [inline ports/cmdline/portfile.cmake:4]:
```suggestion
    REF e4cd007fb8f0314002d9a5b4d82939106e4144e4
```

File: ports/cinatra/vcpkg.json [hunk 10]
```diff
@@ -0,0 +1,12 @@
+{
+    "name": "cinatra",
+    "version-string": "0.9.1",
+    "description": "Cinatra is a high performance HTTP framework with modern C++ features.",
+    "homepage": "https://github.com/qicosmos/cinatra",
```
Comment (LilyWangLL) [inline ports/cinatra/vcpkg.json:5]:
```suggestion
    "homepage": "https://github.com/qicosmos/cinatra",
    "license": "MIT",
```

**Comment by LilyWangLL** [comment]
There is a port which name is [cmdline](https://aur.archlinux.org/packages/cmdline), please add the branch name before the port name, like `tanakh-cmdline`.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by LilyWangLL** [comment]
Can you explain what the following port cinatra does? Why after the installation is complete, this port only installs two tools and puts them under folder include?
image

**Comment by EmployeeNo427** [comment]
> Can you explain what the following port cinatra does? Why after the installation is complete, this port only installs two tools and puts them under folder include? image

Sorry, there are some mistakes on my end in writing the portfile, and this is not the expected behavior; the library has not been installed correctly.

**Comment by LilyWangLL** [comment]
> Please fix the CI pipeline error as below:
> 
> ```
> ##[error]versions/c-/cinatra.json(2,): error : cinatra is missing a version database file at version/c-/cinatra.json
> Run:
> vcpkg x-add-version cinatra
> to create the versions file.
> ##[error]versions/c-/cmdline.json(2,): error : cmdline is missing a version database file at version/c-/cmdline.json
> Run:
> vcpkg x-add-version cmdline
> to create the versions file.
> ```

After you commit the new change, please fix CI errpr by the above solution.

**Comment by EmployeeNo427** [comment]
Hi! Thank you very much for your help! I encountered some issues with my setup that required me to start from scratch. I have opened a new PR for the issue:
[cinatra, cmdline] Add new ports #38712

**Pull request closed without merging**